modules/ROOT/pages/creating-connected-apps-dev.adoc

```diff
@@ -0,0 +1,195 @@
+. In *Website URL* , enter the URL where users go to learn more about the app.
+. In *Redirect URIs*, configure which URIs users are directed to after authorization. 
+.. Type the URL and click *Add* to add it to the list.
+. In *Audience*, select either:
+* *Members of this organization only* if the app is internal to your organization, or
+* *All Anypoint Platform users* if it's a third-party public app that can be distributed outside of your organization.
+include::partial$include-add-scopes.adoc[]
+
+[[create-connected-app-on-its-own-behalf]]
+== Create a Connected App That Acts on Its Own Behalf
+
+include::partial$include-create-connected-apps.adoc[]
+
+. Select *App acts on it's own behalf (client credentials)*.
+include::partial$include-add-scopes.adoc[]
+
+[[grant-types-and-token-flow]]
+== Grant Types and Token Flow
+
+The OAuth framework specification describes grant types for getting access tokens for different use cases.
+
+Anypoint Platform supports these grant types:
+
+* <<auth-code>> 
+* <<refresh-token>>
+* <<password>>
+* <<jwt-bearer>>
+* <<client-credentials>>
+
+For more information and examples on how to use each grant type, see the https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/access-management-api/minor/1.0/pages/Connected%20App%20Examples/[Connected App Examples] in the Access Management API.
+
+[[auth-code]]
+=== Authorization Code
+
+Use the `authorization_code` grant type to authorize the app to access a user's data. 
+
+This is an example of the development flow: 
+
+. A developer creates the connected app that acts on behalf of a user with the `authorization_code` grant type.
+. The end user authorizes the app to act on their behalf.
+. The authorization server sends a temporary authorization code back to the specified redirect URI.
+. The app calls the authorization server with the temporary code to receive the access token. +
+The access token can now be used to call Anypoint Platform APIs on the user’s behalf.
+
+[[refresh-token]]
+=== Refresh Token
+
+Use the `refresh_token` grant type to grant access to user data when the user isn’t signed in. Refresh tokens expire after 90 days. Access tokens expire after the user’s session expires.
+
+To work with OpenID Connect, apps have to exchange refresh tokens for new access tokens after the initial authorization to continue accessing user data.
+
+If the app is using the authorization code grant type, the `offline_access` scope is required to use refresh tokens after the original access tokens expire.
+
+To request a new access token, send a request to the token endpoint using the `refresh_token` grant type. When authentication is successful, a refresh token is exchanged for an access token.
+
+[[password]]
+=== Password
+
+Use the `password` grant type when the application exchanges the user’s username and password for an access token.
+
+[[NOTE]]
```

Comment:
Does this formatting work? [[NOTE]] I usually use just one set of brackets.

Reply:
Auto pilot error